reactor-core/src/test/java/reactor/util/context/ContextTest.java

```diff
@@ -464,6 +465,61 @@ void defaultPutAllMap() {
 				.containsEntry(13, "A13");
 	}
 
+	@Test
+	void defaultPutAllHasSameKey() {
+		Map<Object, Object> leftMap = new HashMap<>();
+		leftMap.put(1, "A");
+		leftMap.put(10, "A10");
+		leftMap.put(11, "A11");
+		leftMap.put(12, "A12");
+		leftMap.put(13, "A13");
+		Map<Object, Object> rightMap = new HashMap<>();
+		rightMap.put(1, "A");
+		rightMap.put(2, "B");
+		rightMap.put(3, "C");
+		rightMap.put(4, "D");
+		rightMap.put(5, "E");
+
+		ContextN left = new ContextN(leftMap);
+		ContextN right = new ContextN(rightMap);
+
+		final Context combined = left.putAll(right.readOnly());
+		assertThat(combined).isInstanceOf(ContextN.class);
+		ContextN combinedN = (ContextN) combined;
+		// If they have the same key, they are merged for the same key.
+		assertThat(combinedN).hasSize(9);
+	}
+
+	@Test
+	void defaultPutAllIntoCheckValue() {
+		int key = 1;
+		String oldValue = "A";
+		String newValue = "B";
+
+		Map<Object, Object> leftMap = new HashMap<>();
+		leftMap.put(key, oldValue);
+		leftMap.put(10, "A10");
+		leftMap.put(11, "A11");
+		leftMap.put(12, "A12");
+		leftMap.put(13, "A13");
+		Map<Object, Object> rightMap = new HashMap<>();
+		rightMap.put(key, newValue);
+		rightMap.put(2, "B");
+		rightMap.put(3, "C");
+		rightMap.put(4, "D");
+		rightMap.put(5, "E");
+
+		ContextN left = new ContextN(leftMap);
+		ContextN right1 = new ContextN(rightMap);
+		Context1 right2 = new Context1(key, newValue);
+
+		final Context combined1 = left.putAllInto(right2);
+		final Context combined2 = left.putAllInto(right1);
+
+		// The result is to have the value of the key included as an argument, if the same key exists.
+		Assertions.assertEquals((String) combined1.get(key), (String) combined2.get(key));
+	}
```

Comment:
Can we also use Mockito constructor instrumentation to see that we allocate an object exactly once?

Reply:
I have checked that the test fails, so I am going to fix the code to make it work. 😭 

I don't know if it can be verified with Mockito's mock or spy, because creating a new ContextN object internally. 
I'll check the requirements more.

Reply:
Mockito's spy() and mock() can't transform the final class, so I used a Proxy. 

And I changed the production code like below.

```
newContext.putAll(base.readOnly()); -> ((CoreContext) base).unsafePutAllInto(newContext);
```

Thanks :)